src/sentry/snuba/metrics/naming_layer/mapping.py

```diff
@@ -54,21 +54,21 @@ def get_mri(external_name: Union[Enum, str]) -> str:
         )
 
 
-def get_public_name_from_mri(internal_name: Union[TransactionMRI, SessionMRI, str]) -> str:
+def get_public_name_from_mri(
+    internal_name: Union[TransactionMRI, SessionMRI, str]
+) -> Optional[str]:
+    """Returns the public name from a MRI if its a builtin metric, None otherwise"""
     if not len(MRI_TO_NAME):
         create_name_mapping_layers()
 
     if isinstance(internal_name, Enum):
         internal_name = internal_name.value
     assert isinstance(internal_name, str)
 
-    try:
-        return MRI_TO_NAME[internal_name]
-    except KeyError:
-        raise InvalidParams(f"Unable to find a mri reverse mapping for '{internal_name}'.")
+    return MRI_TO_NAME.get(internal_name)
 
 
-def get_operation_with_public_name(operation: Optional[str], metric_mri: str) -> str:
+def get_operation_with_public_name(operation: Optional[str], metric_mri: str) -> Optional[str]:
```

Comment:
There is an underlying assumption that `get_public_name_from_mri` only returns None in the case of custom measurements, which is not necessarily true because it would also return `None` for privately ingested metrics that we want to hide, and hence don't have a mapping in the naming layer ie `public.py`